framework/src/utils/InputParameters.C

```diff
@@ -927,8 +941,8 @@ InputParameters::setParamHelper<PostprocessorName, Real>(const std::string & nam
                                                          const Real & r_value)
 {
   // Store the default value
-  _params[name]._default_postprocessor_val = r_value;
-  _params[name]._have_default_postprocessor_val = true;
+  _params[name]._default_postprocessor_val[0] = r_value;
```

Comment:
Let's thrown in an assertion (or two) at the top of this routine to make sure that the vector contains at least one element. I believe your sizing routine is only called through parsing right now but there are other ways to set parameters.

Reply:
`setParamHelper` is used in `InputParameters::addParam`. What is its purpose there?
I didn't really understand the point of the function so I simply did the easiest thing.
Shouldn't there be a better way?

Reply:
The helper is there so that we can specialize it based on the types. Specifically it allows you to set one type with another type (two template arguments). That's why we don't just set the the parameter directly in `addParam`. The assert would be used to just protect us from refactoring and breaking something.